### PR TITLE
Ensure availability of base Python environment

### LINKED ISSUE #72: Python_Toolkit: Use requirements.txt file to build PythonEnvironment object
#### Description:

Rather than having to modify the C# code, we should instead reference a requirements.txt file placed in the `./Engine/Python` directory instead - this would also make creation of environments predefined by other codebases replicable using other means like virtualenv.

### LINKED ISSUE #73: Ensure installation of the "base" environment with all toolkit-specific environments
#### Description:

When installing a toolkit-specific environment, some methods may be referenced from the base environment (things like logging, conversions to/from BHoM objects), and as such being able to access these is essential for the downstream toolkit environment to work fully. When installing the toolkit-environment, a check should be made to ensure that the base-environment also exists, and if not - create it!

### PR DESCRIPTION
**Pull request by tg359**
### Issues addressed by this PR

Closes #73 
Closes #72 

Enables creation of requirements.txt file, rather than relying on one for environment creation, and moves definition of the environment into a toolkit-specific *_Toolkit.json for placement in %ProgramData%\BHoM\Settings\Python. Checks to see whether the base Python environment is available for future dependence on core decorators referenced by downstream methods.

### Test files

[BHoM_Test.zip](https://github.com/BHoM/Python_Toolkit/files/8049461/BHoM_Test.zip)

### CODE REVIEW

**Comment by tg359** [comment]
@BHoMBot check required

File: Python_Engine/Compute/InstallPythonEnvironment.cs [hunk 1]
```diff
@@ -25,32 +25,128 @@
...
+
+        }
 
-            PythonEnvironment pythonEnvironment = Create.PythonEnvironment(Query.ToolkitName(), version, packages);
+        [Description("Install the default BHoM Python_Toolkit environment.")]
+        [Input("run", "Set to True to run the PythonEnvironment installer.")]
+        [Input("force", "If the environment already exists recreate it rather than re-using it.")]
+        [Input("configJSON", "Path to a config JSON containing Python environment configuration.")]
+        [Output("pythonEnvironment", "The default BHoM Python_Toolkit environment object.")]
+        public static PythonEnvironment InstallPythonEnvironment(bool run = false, bool force = false, string configJSON = @"C:\ProgramData\BHoM\Settings\Python\Python_Toolkit.json")
```
Comment (FraserGreenroyd) [inline Python_Engine/Compute/InstallPythonEnvironment.cs:142]:
```suggestion
        [PreviousVersion("5.1", "BH.Engine.Python.Compute.InstallPythonEnvironment(System.Boolean, System.Boolean)")]
        public static PythonEnvironment InstallPythonEnvironment(bool run = false, bool force = false, string configJSON = @"C:\ProgramData\BHoM\Settings\Python\Python_Toolkit.json")
```

File: Python_Engine/Compute/InstallPythonEnvironment.cs [hunk 2]
```diff
@@ -25,32 +25,128 @@
...
-        [Description("Install the default BHoM Python_Toolkit environment.")]
-        [Input("run", "Set to True to run the PythonEnvironment installer.")]
+        [Description("Install the BHoM Python environment.")]
+        [Input("pythonEnvironment", "A BHoM Python environment.")]
         [Input("force", "If the environment already exists recreate it rather than re-using it.")]
-        [Output("pythonEnvironment", "The default BHoM Python_Toolkit environment object.")]
-        public static PythonEnvironment InstallPythonEnvironment(bool run = false, bool force = false)
+        [Input("run", "Set to True to run the PythonEnvironment installer.")]
+        [Output("pythonEnvironment", "A BHoM PythonEnvironment object.")]
+        public static PythonEnvironment InstallPythonEnvironment(this PythonEnvironment pythonEnvironment, bool force = false, bool run = false)
```
Comment (FraserGreenroyd) [inline Python_Engine/Compute/InstallPythonEnvironment.cs:41]:
```suggestion
        [PreviousVersion("5.1", "BH.Engine.Python.Compute.InstallToolkitPythonEnvironment(BH.oM.Python.PythonEnvironment, System.Boolean, System.Boolean)")]
        public static PythonEnvironment InstallPythonEnvironment(this PythonEnvironment pythonEnvironment, bool force = false, bool run = false)
```

File: Python_Engine/Create/PythonEnvironment.cs [hunk 3]
```diff
@@ -60,6 +61,42 @@ public static PythonEnvironment PythonEnvironment(string name, PythonVersion ver
                 Packages = packages,
             };
         }
+
+        [Description("Create a BHoM Python environment from an environment.json config file.")]
+        [Input("config", "The path to the environment.json config file.")]
```
Comment (FraserGreenroyd) [inline Python_Engine/Create/PythonEnvironment.cs:66]:
```suggestion
        [Input("configJSON", "The path to the environment.json config file.")]
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot fix project file ref. 5214476667

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check null-handling
@BHoMBot check serialisation

**Review by FraserGreenroyd — APPROVED** [review]
Have run through with @tg359 and this is good to go to alpha for further testing.

**Pull request merged**